### PR TITLE
Support larger impact graphs

### PR DESCRIPTION
**Pull request by cconard96**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -

Currently, the `positions` field in `glpi_impactcontexts` uses TEXT which can store around 16,383 4-byte characters and each item on the graph seems to take around 60 characters in the JSON (mostly from using 13 decimal places for each coordinate value), leaving room for around 270 items total. In large/complex environments, it may be desired to view more devices all in the same graph.
MEDIUMTEXT would support around 69,905 items in each graph.

### CODE REVIEW

File: inc/migration.class.php [hunk 1]
```diff
@@ -325,6 +325,17 @@ private function fieldFormat($type, $default_value, $nodefault = false) {
             }
             break;
 
+         case 'mediumtext' :
```
Comment (cedric-anne) [inline inc/migration.class.php:328]: We may factorize text/medium/longtext cases, like this:
```php
            $format = sprintf('%s collate %s', strtoupper($type), $collate);
            if (!$nodefault) {
               if (is_null($default_value)) {
                  $format .= " DEFAULT NULL";
               } else {
                  $format .= " NOT NULL DEFAULT '$default_value'";
               }
            }
```

**Comment by cconard96** [comment]
I don't know why tests are failing now in GitHub when they pass locally. I don't even know why the bad schema mentions `longtext` when that type isn't used by me anywhere.

File: install/migrations/update_9.5.x_to_10.0.0/impact.php [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+ */
+/**
+ * @var DB $DB
+ * @var Migration $migration
+ */
+
+/** Impact Relations improvements */
+
+$migration->changeField('glpi_impactcontexts', 'positions', 'positions', 'mediumtext', [
+   'after' => 'id',
```
Comment (trasher) [inline install/migrations/update_9.5.x_to_10.0.0/impact.php:40]:
```suggestion
```
This does not seems relevant

File: inc/migration.class.php [hunk 3]
```diff
@@ -315,23 +315,18 @@ private function fieldFormat($type, $default_value, $nodefault = false) {
...
-               if (is_null($default_value)) {
-                  $format.= " DEFAULT NULL";
-               } else {
-                  $format.= " NOT NULL DEFAULT '$default_value'";
-               }
-            }
-            break;
-
+         case 'mediumtext' :
          case 'longtext' :
```
Comment (trasher) [inline inc/migration.class.php:319]: Did'nt you miss the `mediumtext` case?

**Comment by cedric-anne** [comment]
> I don't know why tests are failing now in GitHub when they pass locally. I don't even know why the bad schema mentions `longtext` when that type isn't used by me anywhere.

During utf8mb4 migration, mediumtext is changed to longtext. I am trying to handle this in the database schema checker.

File: inc/migration.class.php [hunk 4]
```diff
@@ -315,23 +315,18 @@ private function fieldFormat($type, $default_value, $nodefault = false) {
...
-            $format = "LONGTEXT COLLATE $collate";
+            $format = sprintf('%s collate %s', strtoupper($type), $collate);
```
Comment (cedric-anne) [inline inc/migration.class.php:320]:
```suggestion
            $format = sprintf('%s COLLATE %s', strtoupper($type), $collate);
```

**Review by cedric-anne — APPROVED** [review]

**Review by orthagh — APPROVED** [review]

**Review by trasher — APPROVED** [review]

**Pull request merged**